dependency-injection/DependencyInjection/Controller.cs

```diff
@@ -3,12 +3,28 @@
 namespace DependencyInjection;
 
 [ApiController]
-public class Controller(Singleton _singleton)
+public class Controller(IServiceProvider _serviceProvider, Singleton _singleton)
 {
     [HttpPost]
     [Route("stuff")]
     public void DoStuff()
     {
         _singleton.DoStuff("controller");
     }
+
+    [HttpPost]
+    [Route("from-keyed-services/personal-a")]
+    public string FromKeyedServicesUsing([FromKeyedServices(ServiceImplementation.PersonalA)] IPersonal personal)
+    {
+        return personal.Name;
+    }
+
+    [HttpPost]
+    [Route("get-required-keyed-service-using/personal-b")]
+    public string GetRequiredKeyedServiceUsing()
+    {
+        var personal = _serviceProvider.GetRequiredKeyedService<IPersonal>(ServiceImplementation.PersonalB);
+
+        return personal.Name;
+    }
```

Comment:
we won't be using service provider and constructor injection. use `FromKeyedServices` here as well